api/node.go

```diff
@@ -58,6 +62,76 @@ func (s *Server) NodeUpdate(rw http.ResponseWriter, req *http.Request) error {
 	}
 
 	unode, err := s.m.UpdateNode(node)
-	apiContext.Write(toNodeResource(unode, nodeIPMap[node.Name]))
+	apiContext.Write(toNodeResource(unode, nodeIPMap[node.Name], apiContext))
+	return nil
+}
+
+func (s *Server) DiskUpdate(rw http.ResponseWriter, req *http.Request) error {
+	var diskUpdate DiskUpdateInput
+	apiContext := api.GetApiContext(req)
+	if err := apiContext.Read(&diskUpdate); err != nil {
+		return err
+	}
+
+	id := mux.Vars(req)["name"]
+	node, err := s.m.GetNode(id)
+	if err != nil {
+		return errors.Wrap(err, "fail to get node")
+	}
+
+	nodeIPMap, err := s.m.GetManagerNodeIPMap()
+	if err != nil {
+		return errors.Wrap(err, "fail to get node ip")
+	}
+
+	diskUpdateMap := map[string]types.DiskSpec{}
+	for _, uDisk := range diskUpdate.Disks {
+		diskUpdateMap[uDisk.Path] = uDisk
+	}
+
```

Comment:
It would be easier if you translate diskUpdate.Disks to a `fsid` to `disk` map first. It's possible that mount/umount happens for the same path so `fsid` may change between the update. So it's not reliable to use the path as the key for processing.

You can also detect the duplicate `fsid` in this way.